pallets/loans/src/functions.rs

```diff
@@ -665,14 +690,34 @@ impl<T: Config> Pallet<T> {
 							write_off_percentage,
 							write_off_penalty_rate,
 							new_write_off_status,
+							previous_interest_rate,
 						) = match action {
 							WriteOffAction::WriteOffToCurrentGroup => {
 								// Loans that were already written off by an admin,
 								// cannot be written off to the current group anymore.
-								let is_written_off_by_admin = match active_loan.write_off_status {
-									WriteOffStatus::WrittenOffByAdmin { .. } => true,
-									_ => false,
-								};
+								let (previous_interest_rate, is_written_off_by_admin) =
```

Comment:
All of this mess is the big driver of the refactor I want to do around computed rates with penalties... This is unacceptably complex code below here (it was already pretty nasty)